scripts/ipxe-dts.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+error_exit() {
+    _error_msg="$1"
+    echo "$_error_msg"
+    exit 1
+}
+
+error_check() {
+    _error_code=$?
+    _error_msg="$1"
+    [ "$_error_code" -ne 0 ] && error_exit "$_error_msg : ($_error_code)"
+}
+
+userIP=$(ip route get 1 | sed 's/^.*src \([^ ]*\).*$/\1/;q')
+error_check "cannot get user IP"
+
+distroVersion=$(cat meta-dts/meta-dts-distro/conf/distro/dts-distro.conf | grep -F "DISTRO_VERSION" | awk -F'"' '{print $2}')
+error_check "cannot get distro version from 'meta-dts/meta-dts-distro/conf/distro/dts-distro.conf'"
+
+cd meta-dts/scripts/
+error_check "cannot cd to 'meta-dts/scripts'"
+
+mkdir -p ipxe-files
+error_check "cannot make directory 'ipxe-files'"
+
+cd ipxe-files
+error_check "cannot cd to 'ipxe-files'"
+
+wget --output-document=bzImage-v$distroVersion http://boot.dasharo.com/dts/v$distroVersion/bzImage-v$distroVersion
+error_check "cannot wget the bzImage v'$distroVersion' from 'http://boot.dasharo.com/dts/v$distroVersion/bzImage-v$distroVersion'"
+
+wget --output-document=dts-base-image-v$distroVersion.cpio.gz http://boot.dasharo.com/dts/v$distroVersion/dts-base-image-v$distroVersion.cpio.gz
+error_check "cannot wget the cpio.gz v'$distroVersion' from 'http://boot.dasharo.com/dts/v$distroVersion/dts-base-image-v$distroVersion.cpio.gz"
+
+if [ "$#"==0 ]; then
+
+    port=8000
+    python3 -m http.server $port &
+    error_check "failed to create http server with python on port:$port, ip:$userIP"
+
+else
+
+    port=$1
+    python3 -m http.server $port &
+    error_check "failed to create http server with python on port:$port, ip:$userIP"
+
+fi
+
+cat <<EOF >dts.ipxe
+#!ipxe
+#
+kernel http://$userIP:$port/bzImage-v$distroVersion root=/dev/nfs initrd=http://$userIP/dts-base-image-v$distroVersion.cpio.gz
+initrd http://$userIP:$port/dts-base-image-v$distroVersion.cpio.gz
+boot
+EOF
+error_check "cannot create 'dts.ipxe' bootchain file"
+
+
+cat <<EOF
```

Comment:
This is not printed as long as we do not close the HTTP server. Please add some help func, here an example https://git.3mdeb.com/3mdeb/meta-otab/src/branch/master/meta-otab-common/recipes-otab/otab/otab/otab#L15 and then provide a menu in similar fashion to https://git.3mdeb.com/3mdeb/meta-otab/src/branch/master/meta-otab-common/recipes-otab/otab/otab/otab#L439 E.g. we could add `serve` option where second argument would be the port number. In such case, given above variable definition should change to `port=${2:-9000}`

Reply:
@TomaszAIR I made it so that this message gets printed before we start the server, that way it does give the user info about what to do/how to boot. This was done [here](https://github.com/Dasharo/meta-dts/pull/96/commits/eb5dd73352f83f419cdf51aef35e169b5f3dc08c#diff-50ccba31fdf15d04ffc9cadbcbbfc4ceebd19e4076523602a1ee758e9a72cc68R56-R65)

Reply:
What gives us printing such information

```
This script by default sets the server up on port 9000. If you want to use a
different port, for example 9001, then run the script like this:
    ./meta-dts/scripts/ipxe-dts.sh 9001
```

After we executed the script? I would rather have some help that tells that.

Reply:
@TomaszAIR i have made the help more visible with the addition of these two lines [here](https://github.com/Dasharo/meta-dts/pull/96/files#diff-50ccba31fdf15d04ffc9cadbcbbfc4ceebd19e4076523602a1ee758e9a72cc68R51) and [below](https://github.com/Dasharo/meta-dts/pull/96/files#diff-50ccba31fdf15d04ffc9cadbcbbfc4ceebd19e4076523602a1ee758e9a72cc68R62). The user can very quickly stop the script and rerun it again with a different port if they wish.

Reply:
This in no way addresses my comment from the review, I can imagine a user that would like to get any help-related information before executing the script. Please, provide a mechanism that I mentioned in first comment of this thread.

Reply:
@TomaszAIR I added a query to the user at the beginning, informing them about port usage and prompting them whether they want to continue or not. It was added in [this commit](https://github.com/Dasharo/meta-dts/pull/96/commits/a8007e1635addc4da7db79aa594622834134951d)